File: ShellGUI/lib/host/host.dart
// ignore_for_file: unused_import, library_prefixes

import '/common.dart';

// ----------------

abstract class Host {

  Future<Void>
  start(
  );

  Future<Void>
  finish(
  );

  Future<List<String>>
  execute(
    List<String> argument,
  );

}

File: Helper/Bridge/Host.cs
#pragma warning disable 0,
// ReSharper disable CheckNamespace

using Helper;
using Helper.Utility;

namespace Helper.Bridge {

	public abstract class Host {

		#region function

		public abstract void Start (
		);

		public abstract void Finish (
		);

		// ----------------

		public abstract List<String> Execute (
			List<String> argument
		);

		#endregion

	}

}
